out_writeapi.go

```diff
@@ -160,6 +164,20 @@ func checkResponses(curr_ctx context.Context, currQueuePointer *[]*managedwriter
 	return 0
 }
 
+type MWManagedStream interface {
+	AppendRows(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error)
+	Close() error
+}
+
+// To inject a mock interface, I override getWriter and getContext
+var getWriter = func(client *managedwriter.Client, ctx context.Context, projectID string, opts ...managedwriter.WriterOption) (MWManagedStream, error) {
+	return client.NewManagedStream(ctx, opts...)
+}
+
+var getContext = func(ctx unsafe.Pointer) int {
```

Comment:
Could we rename this to getFLBPluginContext()? The term "context" is heavily used in go.

Reply:
Done.